crates/mir/src/body.rs

```diff
@@ -191,9 +191,10 @@ impl<'a> FunctionLower<'a> {
                 hir::Symbol::Param { name: _, param } => LoweredExpr::Operand(Operand::Place(
                     Place::Param(self.get_param_by_expr(*param)),
                 )),
-                hir::Symbol::Local { name: _, expr } => LoweredExpr::Operand(Operand::Place(
-                    Place::Local(self.get_local_by_expr(*expr)),
-                )),
+                hir::Symbol::Local { binding, .. } => {
+                    let expr = binding.lookup(self.hir_file.db(self.db)).expr;
+                    LoweredExpr::Operand(Operand::Place(Place::Local(self.get_local_by_expr(expr))))
+                }
```

Comment:
**Code Structure Issue**: Refactored code for handling local symbol binding lacks clarity and may introduce errors if binding lookup fails. <br> **Fix**: Add error handling for the case where 'binding.lookup' might return None. <br> **Code Suggestion**: 
 ```
 +                hir::Symbol::Local { binding, .. } => {
 +                    let expr = binding.lookup(self.hir_file.db(self.db)).expr;
 +                    LoweredExpr::Operand(Operand::Place(Place::Local(self.get_local_by_expr(expr))))
 +                }
 ```